packages/api/src/dataProviders/db/paymentFlat/paymentFlat.port.ts

```diff
@@ -87,8 +87,22 @@ export class PaymentFlatPort extends MongoPort<Omit<PaymentFlatDbo, "_id">> {
     public async findBySiren(siren: Siren) {
         return this.collection
             .find({
-                typeIdEntrepriseBeneficiaire: "siren",
-                idEntrepriseBeneficiaire: new RegExp(`^${siren.value}\\d{5}`),
+                $expr: {
+                    $or: [
+                        {
+                            typeIdEntrepriseBeneficiaire: "siren",
+                            idEntrepriseBeneficiaire: siren.value,
+                        },
+                        {
+                            $and: [
+                                {
+                                    typeIdEtablissementBeneficiaire: "siret",
+                                    idEntrepriseBeneficiaire: new RegExp(`^${siren.value}\\d{5}`),
+                                },
+                            ],
+                        },
+                    ],
+                },
```

Comment:
J'ai fait ça en passant parce que le code initial était pas bon mais la regex ça doit ralentir je sais pas si on veut garder ça

Reply:
pourquoi on a un $and au deuxième item de $or ? il manque pas un truc dans le premier object concernant le siren ?

Reply:
Non le `$and` est sous-entendu mais je l'ai explicité parce que je pensais avoir besoin pour la regex mais je crois qu'en fait non. Je vais simplifier